org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/internal/parser/XMLParserTest.java

```diff
@@ -8,79 +8,282 @@
-				"[{ tag: 'title', start: 0, end: 20, endTagStart: 12, closed: true, children: [{ tag: 'div', start: 7, end: 12, endTagStart: null, closed: false, children: [] }] }]");
-		assertDocument("<h1><div><span></h1>",
-				"[{ tag: 'h1', start: 0, end: 20, endTagStart: 15, closed: true, children: [{ tag: 'div', start: 4, end: 15, endTagStart: null, closed: false, children: [{ tag: 'span', start: 9, end: 15, endTagStart: null, closed: false, children: [] }] }] }]");
+	public void testCDATABasicNotClosed() {
+		Node text = createNode("testText]</div>", 5, -1, 29, false, true); 
+		Node div = createNode("div", 0, -1, 29, false, false, text);
+
+		assertDocument("<div><![CDATA[testText]</div>", div);
 	}
 
-	private static void assertDocument(String input, String expected) {
-		XMLDocument document = XMLParser.getInstance().parse(input, null);
-		Assert.assertEquals(expected, toJSON(document.getRoots()));
+
+
+
+
+
+
+
+	//--------------------------------------------------------------------------------
+	//Tools
+
+	// public Node createNode(String tag, int start, int endTagStart, int end , boolean closed, boolean isCDATA) {
+	// 	Node n = new Node(start, end, new ArrayList<>(), null, null);
+	// 	setRestOfNode(n, tag, endTagStart, closed, isCDATA);
+	// 	return n;
+	// }
+
+	public Node createNode(String tag, int start, int endTagStart, int end , boolean closed, boolean isCDATA, Node ... children) {
+		ArrayList<Node> newChildren = new ArrayList<Node>(Arrays.asList(children));
+		Node n = new Node(start, end, newChildren, null, null);
+		setRestOfNode(n, tag, endTagStart, closed, isCDATA);
+		return n;
+	}
+
+	// public Node createNode(String tag, int start, int endTagStart, int end , boolean closed, boolean isCDATA, Node child) {
+	// 	return createNode(tag, start, endTagStart, end, closed, isCDATA, Collections.singletonList(child));
+	// }
+
+	private void setRestOfNode(Node n, String tag, int endTagStart, boolean closed, boolean isCDATA) {
+		n.tag = tag;
+		n.endTagStart = endTagStart >= 0 ? new Integer(endTagStart) : null;
+		n.closed = closed;
+		n.isCDATA = isCDATA;
+	} 
+
+	private static void assertDocument(String input, Node currNode) {
+		XMLDocument document = XMLParser.getInstance().parse(input);
+		Node inputRoot = document.children.get(0);
+		compareTrees(currNode, inputRoot);
 	}
 
-	private static String toJSON(Node node) {
-		return "{ tag: '" + node.tag + "', start: " + node.start + ", end: " + node.end + ", endTagStart: "
-				+ node.endTagStart + ", closed: " + node.closed + ", children: " + toJSON(node.children) + " }";
+	private static void assertFailedDocument(String input) {
+		Assert.assertEquals(0, XMLParser.getInstance().parse(input).children.size());
 	}
 
-	private static String toJSON(List<Node> nodes) {
-		return "[" + nodes.stream().map(XMLParserTest::toJSON).collect(Collectors.joining(", ")) + "]";
+	private static void compareTrees(Node root, Node inputRoot){
+		Assert.assertEquals(root.tag, inputRoot.tag);

```

Comment:
do `import static org.junit.Assert.assertEquals;` on top to avoid repeating Assert.assertEquals all over